### DISCUSSION TITLE
LoRA made with "sliders" causes warning

### DISCUSSION THREAD

**Discussion by lbeltrame** [vladmandic/sdnext#2630] [Q&A]
First and foremost, I'm posting here before filing an issue to make sure it *is* an actual issue.
I have been trying to test this LoRA (https://civitai.com/models/228974/great-lighting-sliders-ntcaixyz) for a bit. It appears to have been made following [this approach](https://github.com/rohitgandikota/sliders). 
However, this appears in the log:

```
2023-12-17 19:57:34,950 | sd | WARNING | networks | LoRA unmatched keys: file=models/Lora/great_lighting.safetensors keys=744
```

I'm not sure, however, if:

- This warning means that the LoRA was **not** actually applied
- This warning does **not** mean the LoRA was actually applied and thus can be ignored.

Can someone tell me if this is unexpected behavior? If so I'll file a proper issue.

**Comment by vladmandic** [comment]
if you run `webui --debug`, you should get a bit more information.

but in short, there are tens of subvariations of lora - each one using some slightly different keys. and loader did not recognize 744 keys inside this lora (i don't know out of how many). which means lora was applied only partially (full - 744 keys).
i have no idea whats the engine behind that lora trainer nor what kind of loras does it produce - the github page does not actually say anything on that topic, but a quick look at their code and it seems this is **c3lier** lora type which i've never heard of before.

best-case, someone needs to make a module for that lora type to be able to load/interpret it fully.

**Comment by lbeltrame** [comment]
I figured it would be the case. I'll rerun things in debug and see if there are more clues. I have no idea what LoRA this is either - in CivitAI it is only mentioned basically by this author (and it is also applied at very high weights - 3 or 4). Apparently it's supported by A1111 but a quick glance at the code didn't tell me *where* it is used.

EDIT: LoRA-C3Lier looks like it is actually *LoCon*, and apparently the naming change occurred last April, https://github.com/kohya-ss/sd-scripts/commit/849bc24d205a35fbe1b2a4063edd7172533c1c01

**Comment by vladmandic** [comment]
@AI-Casanova any chance you could take a look? if *c3lier* is basically a *locon*, it should work, but keys do not match.

**Comment by lbeltrame** [comment]
@vladmandic Do you want me to turn this into an issue so it doesn't fall off the radar?

**Comment by AI-Casanova** [comment]
c3lier should be LoCon yes, but sliders did allow more key types than Kohya

Thankfully with a linked problematic LoRA I can dig into it more.

**Comment by lbeltrame** [comment]
I assume this is related to https://github.com/vladmandic/automatic/issues/2868 ?

**Reply by vladmandic** [reply]
yes, opened a feature request for it so it doesn't fall off the radar.

**Comment by lbeltrame** [comment]
The issue has been closed as fixed so this can be closed as well.
